test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/BookkeeperK8sService.java

```diff
@@ -81,6 +81,22 @@ public boolean isRunning() {
                                               t -> new TestFrameworkException(RequestFailed, "Failed to fetch ServiceDetails for bookkeeper", t));
     }
 
+    protected Map<String, Object> patchSpecBuilder(int newInstanceCount) {
+
+        final Map<String, Object> bookkeeperSpec = ImmutableMap.<String, Object>builder()
+                .put("replicas", newInstanceCount)
+                .build();
+
+        return ImmutableMap.<String, Object>builder()
+                .put("apiVersion", "pravega.pravega.io/v1alpha1")
+                .put("kind", CUSTOM_RESOURCE_KIND_PRAVEGA)
+                .put("metadata", ImmutableMap.of("name", PRAVEGA_ID, "namespace", NAMESPACE))
+                .put("spec", ImmutableMap.builder()
```

Comment:
Can you elaborate on what has changed in the lastest `scaleService()` logic here ?

Reply:
Earlier in `scaleService()` all components in the spec were rewritten. As the test pod didn’t contain the arguments passed while running the test, the operator redeployed the components with the default values.
Now only those fields are mentioned in the spec whose values need to be patched, other values remain same as were specified at the time of deployment.